benchmark-framework/src/main/java/io/openmessaging/benchmark/WorkloadGenerator.java

```diff
@@ -109,11 +109,12 @@ public TestResult run() throws Exception {
         producerWorkAssignment.publishRate = targetPublishRate;
         producerWorkAssignment.payloadData = payloadReader.load(workload.payloadFile);
 
-        log.info("----- Starting warm-up traffic ------");
-
         worker.startLoad(producerWorkAssignment);
```

Comment:
Should this be moved inside the if as well?

Reply:
I think you are referring to the `startLoad` call.  I think in the OMB lifecycle `startLoad happens` once. The producers/consumer don't actually know about the warm up/benchmark phases.  This is a coordinator only concern.  The coordinator `resetStats()` after the warm up/backlog building ready for the benchmark phase.